File: python/bubblesort.py
def bubbleSort(arr):
	n = len(arr)
	swapping = False
	
	for i in range(n-1):
		for j in range(0, n-i-1):
			if arr[j] > arr[j + 1]:
				swapped = True
				arr[j], arr[j + 1] = arr[j + 1], arr[j]
		
		if(!swapped):
			return


# Driver code
arr = [10, 57, 85, 13, 2, 191, 70]

bubbleSort(arr)

print("Sorted array is:")
for i in range(len(arr)):
	print("% d" % arr[i], end=" ")

File: java/BubbleSort.java
class BubbleSort {
    

    public static void main(String[] args) {
        int arr[] = { 7, 6, 5, 4, 3, 2, 1 };
        bubblesort(arr);
        for(int i:arr){
            System.out.print(i+" ");
        }
    }

    private static void bubblesort(int[] arr) {
        for(int i=0;i<arr.length-1;i++){
            for(int j=0;j<arr.length-i-1;j++){
                if(arr[j]>arr[j+1]){
                    int temp=arr[j];
                    arr[j]= arr[j+1];
                    arr[j+1]=temp;
                }
            }
        }
    }

}
